Userland/Libraries/LibGUI/Object.h

```diff
@@ -103,155 +103,124 @@ class Object : public Core::EventReceiver {
 
 }
 
+// FIXME: Replace with an analog of std::bind.
+#define BIND_THIS(fn) [this]<typename... Args>(Args&&... args) { return this->fn(forward<Args>(args)...); }
+
 #define REGISTER_INT_PROPERTY(property_name, getter, setter) \
     register_property(                                       \
-        property_name,                                       \
-        [this] { return this->getter(); },                   \
-        [this](auto& value) {                                \
-            this->setter(value.template to_number<int>());   \
-            return true;                                     \
-        });
+        property_name##sv,                                   \
+        BIND_THIS(getter),                                   \
```

Comment:
Not sure I'm thrilled with the extensive use of macros in this patch. especially this one. why is this macro better than `[this] { return this->getter(); },` ?

The suggestion to make an analog of std::bind is confusing. In my experience, every use of std::bind is more readable and more easily understood as either a lambda, or that new fangled `std::bind_front` thing from c++23.